api/src/com/cloud/naming/ResourceNamingPolicyManager.java

```diff
@@ -0,0 +1,32 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.naming;
+
+import com.cloud.utils.component.Manager;
+
+public interface ResourceNamingPolicyManager extends Manager {
+
+
+    /**
+     * Finds a resource naming policy among the registered ones by its class
+     * @param policyClass
+     * @return
+     */
+    @SuppressWarnings("rawtypes")
+    public <T extends ResourceNamingPolicy> T getPolicy(Class<T> policyClass);
```

Comment:
At the end, to me it seems that what you create here is a way to create and manage our own singletons with our own solution, right?

Please do not take me in a bad way, but I really believe we should avoid this kind of solution if can use a simpler solution that is the base of the framework we already use. To me, this looks similar to what has been done to the “JPA” and “Webservice/Rest API” of ACS.